### PR TITLE
docs: align delete range RFC with RocksDB

### PR DESCRIPTION
**Pull request by ben1009**
## Summary

Updates RFC 010 to align the DeleteRange design more closely with RocksDB's range tombstone model.

## Changes

- Add an explicit RocksDB model section for DeleteRange semantics.
- Persist SST range tombstone metadata as fragmented records.
- Preserve all covering tombstone timestamps in fragments for MVCC snapshot correctness.
- Clarify WAL atomicity, recovery, range-only SST invariants, immutable memtable views, scan integration, compaction retention, and vLog GC safety.
- Add rollout phase gates and tests for fragmenter, scan, and compaction edge cases.

## Verification

- cargo make check-typos
- git diff --check
- ./dev check (fmt, dep-sort, clippy, machete, and 423 tests passed; failed in benchmark targets because 27 kv-engine::bench/vlog_benchmarks cases timed out under nextest's 30s limit)

## Summary by CodeRabbit

* **New Features**
  * Added `delete_range(start, end)` with consistent snapshot/MVCC visibility using range tombstones.
  * Enabled range-tombstone aware reads (get/scan/prefix scan) and range-only table support when available.
* **Bug Fixes**
  * Improved correctness for range tombstone handling across WAL recovery and compaction, including safe range-aware GC.
* **Observability**
  * Added range-tombstone statistics to monitor deletion impact.
* **Documentation**
  * Updated RFC 010 with public API, on-disk behavior, and compaction/read semantics.

### CODE REVIEW

File: rfcs/010-delete-range.md [hunk 1]
```diff
@@ -240,14 +271,21 @@ impl LsmStorageInner {
...
-6. Publish the commit timestamp only after the tombstone is visible to readers.
-
-Errors are atomic: if validation, WAL append, or memtable insertion fails, the
-range tombstone is not published and no reader may observe a read timestamp that
-includes it.
+3. Pre-reserve all memory needed for memtable/index insertion.
+4. Reserve one commit timestamp without publishing it to new readers.
+5. Append the range tombstone to the WAL when WAL is enabled.
+6. Insert the range tombstone into the active memtable using only infallible
+   operations backed by the pre-reserved capacity.
```
[the remark this answers is not in the document]
Comment (ben1009) [reply rfcs/010-delete-range.md]: Fixed. Removed "infallible operations backed by pre-reserved capacity" language from Sections 7.1 and 11.1. The RFC now acknowledges that `crossbeam_skiplist::SkipMap` does not support pre-allocation or infallible insertion. After a durable WAL append succeeds, an OOM during in-memory publish should abort the engine rather than leave a committed tombstone unpublished. See commit 5af6331.

File: rfcs/010-delete-range.md [hunk 2]
```diff
@@ -414,9 +456,34 @@ repeated:
...
+records in each table.
+
+The fragmented view consists of non-overlapping spans:
+
+```rust
+pub struct RangeTombstoneFragment {
+    pub start: Bytes,
+    pub end: Bytes,
+    pub covering_ts: Vec<u64>,
+}
```
[the remark this answers is not in the document]
Comment (ben1009) [reply rfcs/010-delete-range.md:543]: Already resolved. The SST block format (lines 480-488) was updated in a prior commit to include `ts_count: u16` followed by `repeated ts: u64`, which matches the `covering_ts: Vec<u64>` field in `RangeTombstoneFragment`. The serialization and in-memory representations are now consistent.

**Comment by ben1009** [comment]
All previous review comments have been addressed and resolved. Please re-review the latest changes.

@gemini-code-assist @coderabbitai
